src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -295,7 +295,11 @@ private void ProcessServerDescriptionChanged(ServerDescriptionChangedEventArgs a
                     {
                         if (newClusterDescription.Type == ClusterType.Unknown)
                         {
-                            newClusterDescription = newClusterDescription.WithType(newServerDescription.Type.ToClusterType());
+                            if (newServerDescription.Type != ServerType.ReplicaSetGhost)
```

Comment:
Which of the new tests fail if you remove this change?

Reply:
https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/134/files#diff-f0a7a201f1347c5dc1a576d47e752ef0R2

Reply:
NOTE: The spec requirements: https://github.com/mongodb/specifications/blob/e56f5eceed7729f8b9b43a4a1f76c7e5840db49f/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologytype-remains-unknown-when-an-rsghost-is-discovered

Reply:
ok, i see. Looks like this was a missing test from the previous version of the spec.